actionpack/lib/action_controller/metal/head.rb

```diff
@@ -38,7 +38,10 @@ def head(status, options = {})
       self.response_body = ""
 
       if include_content?(response_code)
-        self.content_type = content_type || (Mime[formats.first] if formats) || Mime[:html]
+        unless self.media_type
```

Comment:
If I only check `content_type`, there could be cases like `; charset=utf-8` which only have charset value. So I think checking `media_type` would be more accurate.